crates/bevy_ecs/src/schedule/config.rs

```diff
@@ -58,7 +58,7 @@ fn new_condition<M>(condition: impl Condition<M>) -> BoxedCondition {
     let condition_system = IntoSystem::into_system(condition);
     assert!(
         condition_system.is_send(),
-        "Condition `{}` accesses thread-local resources. This is not currently supported.",
+        "Condition `{}` accesses `NonSend` resources. This is not currently supported.",
         condition_system.name()
     );
```

Comment:
I want to note that this assert currently does nothing, since a system's "send-ness" only gets set when it's initialized. This assert never gets triggered, so instead a panic occurs later on when the system actually tries to access a `NonSend` resource from the wrong thread. I'm not sure what the best solution to this is, but it's probably worth opening an issue about it.